docs/src/stake-pool.md

```diff
@@ -1,11 +1,427 @@
+
+```sh
+$ spl-stake-pool remove-validator-stake --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC --stake CrStLEWfme37kDc3nubK9HsmWR5dsuVUuqEKqTR4Mc5E --burn-from 34XMHa3JUPv46ftU4dGHvemZ9oKVjnciRePYMcX3rjEF
+Signature: 5rrQ3xhDWyiPkUTAQkNAeq31n6sMf1xsg2x9hVY8Vj1NonwBnhxuTv87nADLkwC8Xzc4CGTNCTX2Vph9esWnXk2d
+```
+
+This operation works just like `withdraw`, in that the stake pool manager provides
+staking derivatives in exchange for an activated stake account. The difference
+is that the validator stake account is totally removed from the stake pool and
+now belongs to the administrator.
+
+We can check the removed stake account:
+
+```sh
+$ solana stake-account CrStLEWfme37kDc3nubK9HsmWR5dsuVUuqEKqTR4Mc5E
+Balance: 1.002282881 SOL
+Rent Exempt Reserve: 0.00228288 SOL
+Delegated Stake: 1.000000001 SOL
+Active Stake: 1.000000001 SOL
+Delegated Vote Account Address: AUCzCaGAGjL3uyjFBtJs7KuJcgQWvNZu1Z2S9G3pw77G
+Stake Authority: 4SnSuUtJGKvk2GYpBwmEsWG53zTurVM8yXGsoiZQyMJn
+Withdraw Authority: 4SnSuUtJGKvk2GYpBwmEsWG53zTurVM8yXGsoiZQyMJn
+```
+
+The SPL token account has been debited to accommodate the removal of SOL from
+the stake pool.
+
+We can also double-check that the stake pool no longer shows the stake account:
+
+```sh
+$ spl-stake-pool list --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC
+Pubkey: FhFft7ArhZZkh6q4ir1JZMYFgXdH6wkT5M5nmDDb1Q13    Vote: 8r1f8mwrUiYdg2Rx9sxTh4M3UAUcCBBrmRA3nxk3Z6Lm 1.002282881 SOL
+Pubkey: E5KBATUd21Dnjnh5sGFw5ngp9kdVXCcAAYMRe2WsVXie    Vote: HJiC8iJ4Sj846SswQuauFJK93UvV6zp3c2T6jzGqzhhz 11.25999747 SOL
+Pubkey: FYQB64aEzSmECvnG8RVvdAXBxRnzrLvcA3R22aGH2hUN    Vote: 2HUKQz7W2nXZSwrdX5RkfS2rLU4j1QZLjdGCHcoUKFh3 3.163134762 SOL
+Total: 15.425415113 SOL
+```
+
+### Example: (Administrator only) Set staking authority
+
+In order to manage the stake accounts more directly, the stake pool owner can 
+set the stake authority of the stake pool's managed accounts.
+
+```sh
+$ spl-stake-pool set-staking-auth --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC --stake-account FYQB64aEzSmECvnG8RVvdAXBxRnzrLvcA3R22aGH2hUN --new-staker 4SnSuUtJGKvk2GYpBwmEsWG53zTurVM8yXGsoiZQyMJn
+Signature: 39N5gkaqXuWm6JPEUWfenKXeG4nSa71p7iHb9zurvdZcsWmbjdmSXwLVYfhAVHWucTY77sJ8SkUNpVpVAhe4eZ53
+```
+
+Now, the new staking authority can perform any normal staking operations,
+including deactivating or re-staking.
+
+Important security note: the stake pool program only gives staking authority to
+the pool owner and always retains withdraw authority. Therefore, a malicious 
+stake pool manager cannot steal funds from the stake pool.
+
+### Example: (Administrator only) Set owner
+
+The stake pool owner may pass their administrator privileges to another account.
+
+```sh
+$ spl-stake-pool --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC --new-owner 4SnSuUtJGKvk2GYpBwmEsWG53zTurVM8yXGsoiZQyMJn
+Signature: 39N5gkaqXuWm6JPEUWfenKXeG4nSa71p7iHb9zurvdZcsWmbjdmSXwLVYfhAVHWucTY77sJ8SkUNpVpVAhe4eZ53
+```
```

Comment:
Aligning this interface with the existing `solana stake-authorize` command would be good.  A single `spl-stake-pool authorize ...` that takes:
* `--new-staker` -> `--new-stake-authority <PUBKEY>`
* `--new-owner` -> `--new-withdraw-authority <PUBKEY>`

Reply:
First one sounds good, but the second one doesn't quite reflect the situation, since it's setting the stake pool manager, who doesn't have withdraw authority over the account.  How about `--new-owner` -> `new-administrator`?